docs/guides/cdk/using-amplify-with-cdk.md

```diff
@@ -0,0 +1,275 @@
+
+### Getting started
+
+This guide will walk you through adding authentication using the Amplify CLI and then building a GraphQL API that references and uses this authentication service as an authentication and authorization provider.
+
+## Creating the Amplify app
+
+The first step will be to initialize an Amplify app and create a new authentication service.
+
+<amplify-callout>
+
+If you already have an authentication service created using the Amplify CLI, you can jump to [Creating the CDK project](#creating-the-cdk-project)
+
+</amplify-callout>
+
+### Initializing the Amplify project:
+
+```sh
+amplify init
+```
+
+### Adding authentication
+
+```sh
+amplify add auth
+
+? Do you want to use default authentication and security configuration? Default configuration 
+? How do you want users to be able to sign in when using your Cognito User Pool? Username
+? Do you want to configure advanced settings? No, I am done. 
+```
+
+### Deploying the Amplify back end
+
+```sh
+amplify push --y
+```
+
+## Creating the CDK project
+
+Next, create a new folder called __cdk-backend__ and initialize a new CDK project within that folder.
+
+```sh
+mkdir cdk-backend
+cd cdk-backend
+cdk init --language=typescript
+```
+
+Next, install the necessary CDK dependencies for AWS AppSync, AWS Lambda, Amazon Cognito, and Amazon DynamoDB:
+
+```sh
+npm install @aws-cdk/aws-appsync @aws-cdk/aws-cognito @aws-cdk/aws-dynamodb @aws-cdk/aws-lambda
+```
+
+### Consuming the authentication service with CDK
+
+Next, open __lib/cdk-backend-stack.ts__ and update with the following code.
+
+<amplify-callout>
+
+Note that `<your-user-pool-id>` will be the user pool ID generated by the Amplify cli and will be available in __src/aws-exports.js__ as the `aws_user_pools_id` property.
+
+</amplify-callout>
+
+```js
+import * as cdk from '@aws-cdk/core';
+import * as appsync from '@aws-cdk/aws-appsync';
+import * as cognito from '@aws-cdk/aws-cognito';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as dynamodb from '@aws-cdk/aws-dynamodb';
+
+export class CdkBackendStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+  
+    /* Consumes the user pool created by Amplify CLI */
+    const userPool = cognito.UserPool.fromUserPoolId(
+      this,
+      'amplify-user-pool',
+      <your-user-pool-id>
+    )
+
+    /* Creates the AppSync API using the User Pool from Amplify */
+    const api = new appsync.GraphqlApi(this, 'Api', {
+      name: 'cdk-blog-api',
+      schema: appsync.Schema.fromAsset('graphql/schema.graphql'),
+      authorizationConfig: {
+        defaultAuthorization: {
+          authorizationType: appsync.AuthorizationType.API_KEY,
+          apiKeyConfig: {
+            expires: cdk.Expiration.after(cdk.Duration.days(365))
```

Comment:
type error 
![image](https://user-images.githubusercontent.com/6764957/98020151-96656d80-1e3d-11eb-8af7-52a4685f2545.png)

Reply:
ok this was caused by the CDK init adding different version than latest: 
![image](https://user-images.githubusercontent.com/6764957/98021804-b85fef80-1e3f-11eb-8c03-5c4f47ee9fa3.png)

Reply:
Yeah I usually upgrade my CLI before initializing a new CDK project, this happens all of the time to me.